frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardPage/PublicOrEmbeddedDashboardPage.tsx

```diff
@@ -54,35 +57,42 @@ export const PublicOrEmbeddedDashboardPage = (props: WithRouterProps) => {
       locale={canWhitelabel ? locale : undefined}
       shouldWaitForLocale
     >
-      <DashboardContextProvider
-        dashboardId={dashboardId}
-        hideParameters={hide_parameters}
-        theme={theme}
-        background={background}
-        bordered={bordered}
-        downloadsEnabled={downloadsEnabled}
-        titled={titled}
-        parameterQueryParams={parameterQueryParams}
-        cardTitled={true}
-        withFooter={true}
-        getClickActionMode={({ question }) => new Mode(question, PublicMode)}
-        navigateToNewCardFromDashboard={null}
-        onError={(error) => {
-          dispatch(setErrorPage(error));
-        }}
-        isDashcardVisible={(dashcard) => !isActionDashCard(dashcard)}
-        dashcardMenu={({ dashcard, result }) =>
-          downloadsEnabled?.results &&
-          isQuestionCard(dashcard.card) &&
-          !!result?.data &&
-          !result?.error && (
-            <PublicOrEmbeddedDashCardMenu result={result} dashcard={dashcard} />
-          )
-        }
-        dashboardActions={DASHBOARD_DISPLAY_ACTIONS}
-      >
-        <PublicOrEmbeddedDashboardPageInner {...props} />
-      </DashboardContextProvider>
+      <EmbeddingEntityContextProvider uuid={uuid ?? null} token={token ?? null}>
```

Comment:
Why is this value initialization different from `PublicOrEmbeddedQuestion` we just pass `uuid` and `token` there directly without having to cast them to `null`.

These 2 components should look somewhat similar to each other.